### ISSUE TITLE
LG Heat Pump - Which Value is shown?

### ISSUE THREAD

**Issue by Trainerbert** [ollo69/ha-smartthinq-sensors#819]
Hey there,

I am using the Integration with my heat pump LG Therma V R290 and it works pretty good.
But I don't understand which value is shown in the auto menu.
In the picture you can see the heat pump shows the value of 17 degress.
But where does it take this value from. It isn't the outdoor temperature and not a value from the heating curve.

![image](https://github.com/user-attachments/assets/74e837d2-9a03-402a-8357-3930a0677828)

**Comment by jurriaancap** [comment]
i think its the target temperature for the outlet which equals to zero in the thinq app
if you increase it in the thinq app , homeassistent will also update to 21 (if you add 5 in the thinqapp)  or 12 if you lower it
(i have the dutch version , here its called doeltemperatuur uitlaat, which i translated)

**Comment by Trainerbert** [comment]
Yeah I think thats true but pretty bad shown then. 17 means zero offset and 12 is -5 and 22 is +5.
I am missing the real water target and outdoor unit temperatur. So I can't really understand what is going on.
Maybe with the Modbus Version it is better?

**Comment by jurriaancap** [comment]
i have seen some yaml code where they offset the values , i tried to do that by going to to developer tools - states
i have changed the state attributes of that entity , 
hvac_modes:
  - "off"
  - auto
  - heat
min_temp: -5
max_temp: 5
target_temp_step: 1
current_temperature: 34
temperature: 0
friendly_name: Warmtepomp therma v
supported_features: 385

hower that does nothing and it keeps referting back. so obviously thats not wheren you can set the offset (or i am doing it wrong , no judgment please, i just started fiddling with it today ;) )

**Comment by Trainerbert** [comment]
Thanks, I will integrate the Modbus solution.

**Comment by jurriaancap** [comment]
hmm i have been poking around,  and found this part in the diagnostic file
              "airState.ess.solarPowerExt": 0.0,
              "airState.tempState.current": 24.5,
              "airState.tempState.hotWaterTempMax": 55.0,
              "airState.miscFuncState.awhpTempSwitch": 1.0,
              "airState.miscFuncState.extraOp": 0.0,
              "airState.2nd.tempState.airTempCoolMin": 16.0,
              "airState.2nd.tempState.target": 17.0,
              "airState.2nd.tempState.waterTempCoolMin": 5.0,
              "airState.tempState.airTempHeatMax": 30.0,

no clue yet how to change it the entity though

**Comment by qemical** [comment]
They set the pump incorrectly. Very classical of installer nowadays. If your pump is entirely for heating floor, you need to specify that the measure of temperature is WATER, and not AIR.

Its in the settings somewhere under USER of FUNCTION. I will take a look at home when I arrive.

**Comment by ollo69** [comment]
I don't exactly know how your devices work (I don't have heat pump), so you should provide example about what you expect compared to what you see on integration and attach integration diagnostic so that I can search for right place where to get right value. Probably I'm just using wrong information!

**Comment by berndthiel** [comment]
> Hey there,
> 
> I am using the Integration with my heat pump LG Therma V R290 and it works pretty good. But I don't understand which value is shown in the auto menu. In the picture you can see the heat pump shows the value of 17 degress. But where does it take this value from. It isn't the outdoor temperature and not a value from the heating curve.
> 
> ![image](https://private-user-images.githubusercontent.com/20670927/378604197-74e837d2-9a03-402a-8357-3930a0677828.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxMTM5OTYsIm5iZiI6MTczNjExMzY5NiwicGF0aCI6Ii8yMDY3MDkyNy8zNzg2MDQxOTctNzRlODM3ZDItOWEwMy00MDJhLTgzNTctMzkzMGEwNjc3ODI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDIxNDgxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1MGUxN2Q0ZjVlYmExYjFhYzhiZmFhZWIxMWZiNzcwYzExYzgzNzZhZWU0MWNhYjY2YzM5ZjViOWRiZDRmYWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DBbCIwXCyb8lPq_T4Me8naoviPSsT-N0NfdDrCPSqB4)

@Trainerbert the 17° is a non-sense value here... correct should be "outside temp" and "target temp" which are relevant for auto mode based your seasonal settings you set on your lg display @ollo69  can you explain us what data this is?

**Comment by ollo69** [comment]
Probably this is value from `airState.tempState.outWaterCurrent` in the payload.
If you download integration diagnostics and open it with a text editor, you can search payload name associated with value 17 and provide me information about payload name that seems provide the expected value, so that I can fix in the integration.

**Comment by qemical** [comment]
> > Hey there,
> > I am using the Integration with my heat pump LG Therma V R290 and it works pretty good. But I don't understand which value is shown in the auto menu. In the picture you can see the heat pump shows the value of 17 degress. But where does it take this value from. It isn't the outdoor temperature and not a value from the heating curve.
> > ![image](https://private-user-images.githubusercontent.com/20670927/378604197-74e837d2-9a03-402a-8357-3930a0677828.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxMTM5OTYsIm5iZiI6MTczNjExMzY5NiwicGF0aCI6Ii8yMDY3MDkyNy8zNzg2MDQxOTctNzRlODM3ZDItOWEwMy00MDJhLTgzNTctMzkzMGEwNjc3ODI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDIxNDgxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1MGUxN2Q0ZjVlYmExYjFhYzhiZmFhZWIxMWZiNzcwYzExYzgzNzZhZWU0MWNhYjY2YzM5ZjViOWRiZDRmYWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DBbCIwXCyb8lPq_T4Me8naoviPSsT-N0NfdDrCPSqB4)
> 
> @Trainerbert the 17° is a non-sense value here... correct should be "outside temp" and "target temp" which are relevant for auto mode based your seasonal settings you set on your lg display @ollo69 can you explain us what data this is?

You need to set it properly in the heat pump panel on your wall. Can you provide a picture of your panel?

**Comment by berndthiel** [comment]
@ollo69 the challenge here is following

LG Heatpump you define a "lowest outside temp" > "target temp" and a "highest outside temp" > "target temp" and in "auto" the "target temp" should change on any change by outside temp

can you share what kind of data you see in debug?

**Comment by berndthiel** [comment]
> > > Hey there,
> > > I am using the Integration with my heat pump LG Therma V R290 and it works pretty good. But I don't understand which value is shown in the auto menu. In the picture you can see the heat pump shows the value of 17 degress. But where does it take this value from. It isn't the outdoor temperature and not a value from the heating curve.
> > > ![image](https://private-user-images.githubusercontent.com/20670927/378604197-74e837d2-9a03-402a-8357-3930a0677828.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxMTM5OTYsIm5iZiI6MTczNjExMzY5NiwicGF0aCI6Ii8yMDY3MDkyNy8zNzg2MDQxOTctNzRlODM3ZDItOWEwMy00MDJhLTgzNTctMzkzMGEwNjc3ODI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDIxNDgxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1MGUxN2Q0ZjVlYmExYjFhYzhiZmFhZWIxMWZiNzcwYzExYzgzNzZhZWU0MWNhYjY2YzM5ZjViOWRiZDRmYWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DBbCIwXCyb8lPq_T4Me8naoviPSsT-N0NfdDrCPSqB4)
> > 
> > 
> > @Trainerbert the 17° is a non-sense value here... correct should be "outside temp" and "target temp" which are relevant for auto mode based your seasonal settings you set on your lg display @ollo69 can you explain us what data this is?
> 
> You need to set it properly in the heat pump panel on your wall. Can you provide a picture of your panel?

@qemical yes we have to findout which data is correct one, cuase in auto the target temp will be changed by the heatpump depend your seasonal temp settings

**Comment by Trainerbert** [comment]
Basically in the display of the heat pump you have 2 settings. The "auto" setting which targets the heat curve and in the auto mode you have a range of -5 to +5. So the fix would simply be if you set the climate table to auto it will have values from -5 to 5.
At the moment if you set it to auto then it shows 17° which is 0, 12° is -5 and 22° is +5.

**Comment by qemical** [comment]
> > > > Hey there,
> > > > I am using the Integration with my heat pump LG Therma V R290 and it works pretty good. But I don't understand which value is shown in the auto menu. In the picture you can see the heat pump shows the value of 17 degress. But where does it take this value from. It isn't the outdoor temperature and not a value from the heating curve.
> > > > ![image](https://private-user-images.githubusercontent.com/20670927/378604197-74e837d2-9a03-402a-8357-3930a0677828.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxMTM5OTYsIm5iZiI6MTczNjExMzY5NiwicGF0aCI6Ii8yMDY3MDkyNy8zNzg2MDQxOTctNzRlODM3ZDItOWEwMy00MDJhLTgzNTctMzkzMGEwNjc3ODI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDIxNDgxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1MGUxN2Q0ZjVlYmExYjFhYzhiZmFhZWIxMWZiNzcwYzExYzgzNzZhZWU0MWNhYjY2YzM5ZjViOWRiZDRmYWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DBbCIwXCyb8lPq_T4Me8naoviPSsT-N0NfdDrCPSqB4)
> > > 
> > > 
> > > @Trainerbert the 17° is a non-sense value here... correct should be "outside temp" and "target temp" which are relevant for auto mode based your seasonal settings you set on your lg display @ollo69 can you explain us what data this is?
> > 
> > 
> > You need to set it properly in the heat pump panel on your wall. Can you provide a picture of your panel?
> 
> @qemical yes we have to findout which data is correct one, cuase in auto the target temp will be changed by the heatpump depend your seasonal temp settings

That is correct, but if its a heating floor by water, I always prefer to get in the panel the Water temperature. rather that air temperature.

@berndthiel ,  btw, did you update to 2025.1.1 and ? If you did, did dissapear your climate entity of the air to water pump?

**Comment by Trainerbert** [comment]
When I set it on "heat" instead of "auto" I see the water target temperature.

**Comment by qemical** [comment]
> When I set it on "heat" instead of "auto" I see the water target temperature. Edit: Here some pics: ![1 (1)](https://private-user-images.githubusercontent.com/20670927/400258361-3c95bf00-fd0d-4eca-b699-a166b9257205.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxMjE3MTksIm5iZiI6MTczNjEyMTQxOSwicGF0aCI6Ii8yMDY3MDkyNy80MDAyNTgzNjEtM2M5NWJmMDAtZmQwZC00ZWNhLWI2OTktYTE2NmI5MjU3MjA1LmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwNVQyMzU2NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNWEzOWIyM2IxMjI4YmNlMTgwODcyODc4MGZkMGM3YjhiOWFmMmMzODQ3ZjkyNTNiODRlNDg1M2ZhYTYwNTMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2nMNXEPxod-NbgdIp4WdKdidKLZgZ16LSu4miDq1V-E) ![1 (4)](https://private-user-images.githubusercontent.com/20670927/400258362-0350ecb6-a600-4513-97f0-ad7fb5e6f1ea.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxMjE3MTksIm5iZiI6MTczNjEyMTQxOSwicGF0aCI6Ii8yMDY3MDkyNy80MDAyNTgzNjItMDM1MGVjYjYtYTYwMC00NTEzLTk3ZjAtYWQ3ZmI1ZTZmMWVhLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwNVQyMzU2NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNzY5ZWNmNGE3ZWIwZjA4YjJmMTljYzRkM2NmZjM3NzcwN2Q1YzUxZDBkYzgzOWJlYmIwZGYzMjhlZTIyNTgwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.vz-TAb3jkMx5XDvxyoKjxHcVwPTj-7CLJEYFoddVym0) ![1 (3)](https://private-user-images.githubusercontent.com/20670927/400258365-3a06c46e-208b-41e2-a877-9e049e40216b.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxMjE3MTksIm5iZiI6MTczNjEyMTQxOSwicGF0aCI6Ii8yMDY3MDkyNy80MDAyNTgzNjUtM2EwNmM0NmUtMjA4Yi00MWUyLWE4NzctOWUwNDllNDAyMTZiLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwNVQyMzU2NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mOWQzMmQ0ZDU4MTA1NzJjZGJkMjZmMDk5OGY0YjRmMmFiYjNlNzE4NDNjNjg2NDU2MTVkMDg1ZjMyYTRlMDU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.M4XA0OULZaCpLgtlK4Jno7s2AxJOQ7RDXBPltloc5zc) ![1 (2)](https://private-user-images.githubusercontent.com/20670927/400258366-cd78e2f7-d4c0-4db9-8c68-ec129f32302e.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxMjE3MTksIm5iZiI6MTczNjEyMTQxOSwicGF0aCI6Ii8yMDY3MDkyNy80MDAyNTgzNjYtY2Q3OGUyZjctZDRjMC00ZGI5LThjNjgtZWMxMjlmMzIzMDJlLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwNVQyMzU2NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zOTI4NGE4NjRhNzI4M2E1NTg0NzEyOTE2NTZkYmVjYzgxYzkwZTNhZmZlNjBlYzE2ODA5NzA4YjYxZjExMTYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pFmipwmt-ZzWVvCPsaWYLwbDtRcfUgZsl4JOvajL_S8)

Great. You have the same panel than me. Unfortunately I am at my parents for vacations and I will be bacl in a one week.

There is setting in the panel where you set if you want Air or Water. But I believe you have to enter into Installer Menu and it asks for a code. Let me rescue the manual I will send to you asap. In the meantime, look what my panel shows

![image](https://github.com/user-attachments/assets/bfb6ddc2-13e8-4336-8296-00780951e6bf)

As you can see in my climate, I can go from 5 to 65. And the 13º degrees shows the current water temp of the circuit.

THose values of 5 to 65 it can be also modify in the panel if I manage to send you the manual.

**Comment by Trainerbert** [comment]
Yeah, I know :). I was in the menu and my last Picture is the setting where you can set it to water, air or water+air. I already use the modbus mod to get more values.

**Comment by qemical** [comment]
Well.... I am not fluent in German. But the min and max temp can set in a menu where you have to insert a pin password.

I'll look for it.

Are you using the USB WiFi modem. How are you integrating the pump?

**Comment by Trainerbert** [comment]
I know, the password is the firmware version. I know literal all settings. Otherwise you couldnt adjust the heat curve or use modbus. It is just a visual problem in ha, nothing else.

**Comment by qemical** [comment]
> I know, the password is the firmware version. I know literal all settings. Otherwise you couldnt adjust the heat curve or use modbus. It is just a visual problem in ha, nothing else. Yes usb wifi modem and also waveshare for modbus.

Exactly. In that case. I have nothing to add here.

Did you experiment some crashes that modem disconnect from wifi and it requires the reboot of the pump? Happens to me many times when I change temperatures. A bit annoying.

**Comment by Trainerbert** [comment]
Yes, like 2 times with the error "ch03".
